node/src/block_producer/vrf_evaluator/block_producer_vrf_evaluator_reducer.rs

```diff
@@ -164,7 +314,40 @@ impl BlockProducerVrfEvaluatorState {
                     next_epoch_first_slot: *next_epoch_first_slot,
                     staking_epoch_data: staking_epoch_data.clone(),
                     producer: producer.clone(),
-                }
+                };
+
+                let (dispatcher, state) = state_context.into_dispatcher_and_state();
+                let get_slot_and_status = || {
+                    let cur_global_slot = state.cur_global_slot()?;
+                    let status = &state.block_producer.vrf_evaluator()?.status;
+
+                    Some((cur_global_slot, status))
+                };
+
+                let Some((
+                    current_global_slot,
+                    BlockProducerVrfEvaluatorStatus::EpochDelegatorTableSuccess {
+                        best_tip_epoch,
+                        best_tip_slot,
+                        best_tip_global_slot,
+                        next_epoch_first_slot,
+                        staking_epoch_data,
+                        ..
+                    },
+                )) = get_slot_and_status()
+                else {
+                    // error here!
```

Comment:
I guess this should be a `bug_condition`. If this fails it means that we don't have a best tip yet or that the vrf evaluator isn't ready (then we shouldn't be handling this action)